### PR TITLE
fix: Fix focus ring styles for radio button toggles #1772

### LINKED ISSUE #1772: Focus ring not shown for radio toggles
Draft work happening in #1773

### Requirements

- [X] This is a bug report, and if not, please post to https://lemmy.ml/c/lemmy_support instead.
- [X] Please [check](https://github.com/LemmyNet/lemmy-ui/issues) to see if this issue already exists.
- [X] It's a single bug. Do not report multiple bugs in one issue.
- [X] It's a frontend issue, not a backend issue; Otherwise please create an issue on the [backend repo](https://github.com/LemmyNet/lemmy) instead.

### Summary

When the "toggle"-style radio buttons are focused, there should be a ring to let the user know they're focused. This doesn't show up, because the `<label>` and `<input>` fields are in the wrong order.

Label needs to be sibling element to input, like this:

```jsx
            <input
              id={`${this.id}-subscribed`}
              type="radio"
              className="btn-check"
              value={"Subscribed"}
              checked={this.state.type_ == "Subscribed"}
              onChange={linkEvent(this, this.handleTypeChange)}
              disabled={!UserService.Instance.myUserInfo}
            />
            <label
              htmlFor={`${this.id}-subscribed`}
              title={I18NextService.i18n.t("subscribed_description")}
              className={`btn btn-outline-secondary 
              ${this.state.type_ == "Subscribed" && "active"}
              ${!UserService.Instance.myUserInfo ? "disabled" : "pointer"}
            `}
            >
              {I18NextService.i18n.t("subscribed")}
            </label>
```

### Steps to Reproduce

1. Tab until the focus is on a radio toggle
2. See that there is no focus ring

### Technical Details

Latest `main`

### Lemmy Instance Version

main

### Lemmy Instance URL

_No response_

### PR DESCRIPTION
**Pull request by jsit**
Fixes #1772 

## Before

Screenshot 2023-07-04 at 3 36 29 PM

## After

Screenshot 2023-07-04 at 3 35 21 PM

### CODE REVIEW

File: src/shared/components/common/data-type-select.tsx [hunk 1]
```diff
@@ -31,33 +35,41 @@ export class DataTypeSelect extends Component<
...
+      <div
+        className="data-type-select btn-group btn-group-toggle flex-wrap"
+        role="group"
+      >
+        <input
+          id={`${this.id}-posts`}
+          type="radio"
+          className="btn-check"
+          value={DataType.Post}
+          checked={this.state.type_ == DataType.Post}
```
Comment (SleeplessOne1917) [inline src/shared/components/common/data-type-select.tsx:47]: Use triple equals.

File: src/shared/components/common/data-type-select.tsx [hunk 2]
```diff
@@ -31,33 +35,41 @@ export class DataTypeSelect extends Component<
...
-          />
           {I18NextService.i18n.t("posts")}
         </label>
+
+        <input
+          id={`${this.id}-comments`}
+          type="radio"
+          className="btn-check"
+          value={DataType.Comment}
+          checked={this.state.type_ == DataType.Comment}
```
Comment (SleeplessOne1917) [inline src/shared/components/common/data-type-select.tsx:64]: Triple equals.

File: src/shared/components/person/registration-applications.tsx [hunk 3]
```diff
@@ -125,34 +127,41 @@ export class RegistrationApplications extends Component<
...
-          />
           {I18NextService.i18n.t("unread")}
         </label>
+
+        <input
+          id={`${radioId}-all`}
+          type="radio"
+          className="btn-check"
+          value={UnreadOrAll.All}
+          checked={this.state.unreadOrAll == UnreadOrAll.All}
```
Comment (SleeplessOne1917) [inline src/shared/components/person/registration-applications.tsx:156]: Triple equals

File: src/shared/components/person/reports.tsx [hunk 4]
```diff
@@ -187,95 +189,107 @@ export class Reports extends Component<any, ReportsState> {
...
+
     return (
-      <div className="btn-group btn-group-toggle flex-wrap mb-2">
+      <div className="btn-group btn-group-toggle flex-wrap mb-2" role="group">
+        <input
+          id={`${radioId}-unread`}
+          type="radio"
+          className="btn-check"
+          value={UnreadOrAll.Unread}
+          checked={this.state.unreadOrAll == UnreadOrAll.Unread}
```
Comment (SleeplessOne1917) [inline src/shared/components/person/reports.tsx:201]: Triple equals.

File: src/shared/components/person/reports.tsx [hunk 5]
```diff
@@ -187,95 +189,107 @@ export class Reports extends Component<any, ReportsState> {
...
-          />
           {I18NextService.i18n.t("unread")}
         </label>
+
+        <input
+          id={`${radioId}-all`}
+          type="radio"
+          className="btn-check"
+          value={UnreadOrAll.All}
+          checked={this.state.unreadOrAll == UnreadOrAll.All}
```
Comment (SleeplessOne1917) [inline src/shared/components/person/reports.tsx:218]: Triple equals

File: src/shared/components/person/reports.tsx [hunk 6]
```diff
@@ -187,95 +189,107 @@ export class Reports extends Component<any, ReportsState> {
...
+
     return (
-      <div className="btn-group btn-group-toggle flex-wrap mb-2">
+      <div className="btn-group btn-group-toggle flex-wrap mb-2" role="group">
+        <input
+          id={`${radioId}-all`}
+          type="radio"
+          className="btn-check"
+          value={MessageType.All}
+          checked={this.state.messageType == MessageType.All}
```
Comment (SleeplessOne1917) [inline src/shared/components/person/reports.tsx:243]: Triple equals

File: src/shared/components/person/reports.tsx [hunk 7]
```diff
@@ -187,95 +189,107 @@ export class Reports extends Component<any, ReportsState> {
...
-          />
           {I18NextService.i18n.t("all")}
         </label>
+
+        <input
+          id={`${radioId}-comments`}
+          type="radio"
+          className="btn-check"
+          value={MessageType.CommentReport}
+          checked={this.state.messageType == MessageType.CommentReport}
```
Comment (SleeplessOne1917) [inline src/shared/components/person/reports.tsx:260]: Triple equals

File: src/shared/components/person/reports.tsx [hunk 8]
```diff
@@ -187,95 +189,107 @@ export class Reports extends Component<any, ReportsState> {
...
-          />
           {I18NextService.i18n.t("comments")}
         </label>
+
+        <input
+          id={`${radioId}-posts`}
+          type="radio"
+          className="btn-check"
+          value={MessageType.PostReport}
+          checked={this.state.messageType == MessageType.PostReport}
```
Comment (SleeplessOne1917) [inline src/shared/components/person/reports.tsx:277]: Triple equals

**Review by SleeplessOne1917 — COMMENTED** [review]
Replace all the double equals with triple equals and you're good.

File: src/shared/components/common/data-type-select.tsx (hunk 1 above)
Comment (jsit) [reply src/shared/components/common/data-type-select.tsx:47]: Yeah there are a loooooot of these everywhere, I was tempted to replace them everywhere in all the files I touched here.

**Review by SleeplessOne1917 — APPROVED** [review]

**Review by SleeplessOne1917 — APPROVED** [review]

**Review by dessalines — APPROVED** [review]
Not sure why lint is failing.

**Pull request merged**